sale_stock_partner_warehouse/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    warehouse_id = fields.Many2one(
+        compute="_compute_warehouse_id", store=True, precompute=True
+    )
+
+    @api.depends("user_id", "company_id", "partner_id.sale_warehouse_id")
```

Comment:
If you change the wh on the partner, you probably don't want to recompute all SO of that partner
```suggestion
    @api.depends("user_id", "company_id", "partner_id")
```